src/mono/CMakeLists.txt

```diff
@@ -848,6 +848,9 @@ endif()
 ### End of OS specific checks
 
 include_directories("${CLR_SRC_NATIVE_DIR}")
+
+add_subdirectory("${CLR_SRC_NATIVE_DIR}/include/mono-2.0" mono-2.0)
```

Comment:
Do we need the version number in the path? What does the version number mean?

We version the whole runtime together. I think trying to support multiple versions in the same repo would be very complex.

Reply:
I'm trying to match the layout that the existing runtime pack uses for the headers.

The mono API is always going to be "2.0" as long as its API/ABI compatible.

Reply:
> I'm trying to match the layout that the existing runtime pack uses for the headers.

Where is the existing runtime pack?

Reply:
https://nuget.info/packages/Microsoft.NETCore.App.Runtime.Mono.iossimulator-x64/6.0.1

The headers are in `runtimes/[RID]/native/include/mono-2.0/...` 

The code that places them there is:

https://github.com/dotnet/runtime/blob/69b9000671f1b73b3fa17810a9f6cb31abb612a2/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.props#L69-L73

(this is also matching the layout that the old mono/mono packages use - `include/mono-2.0/mono/{utils,metadata,jit}/...`)

Reply:
@vargaz @jkotas  renamed the directory to `src/native/monoapi` in the source tree (so headers are `src/native/monoapi/mono/metadata/object.h` etc).  Still installed as `mono-2.0` in the runtime pack.

Reply:
Perhaps there are still too many mono names. Runtime pack includes Mono in name, then you have monoapi and mono under it.

Reply:
I'm not sure how much flexibility I have with the names.

In the runtime, we typically have includes like `#include <mono/metadata/object.h>`  so at least one level of `mono/` has to be in the repo.  In embedding code, the `mono-2.0/mono/metadata/object.h` directory layout is pretty well established and embedders also typically write `#include <mono/metadata/object.h>.

So if I move these headers, in the repo, they need to go some place like `src/native/XYZ` and beneath that it seems like we must have `mono/metadata/object.h`.

We could do something like`src/native/include/mono/metadata/object.h` but I don't think that conveys the idea that these headers are not an internal implementation detail of the runtime, but are part of the public API.

What about `src/native/public/mono/metadata/object.h`?  or even `src/public/mono/metadata/object.h` ?

Reply:
> If the idea is that all public headers that are designed to be included from outside the repo should live in a central place separate from their implementation, we should think about the directory structure for all of these.

Yea an implementation-independent home for nethost.h and hostfxr.h would be great. (And would make it a bit easier for the mobile hosts to implement the same APIs)

I'm not sure what kind of directory layout would be best.  `src/native` seems more natural than a new toplevel folder under `src/` since these are native APIs. But on the other hand `src/native` is getting crowded.

Reply:
Personally, I like `src/native/public` (e.g. src/native/public/mono/metadata/object.h) that you have proposed above.